src/startupscriptgenerator/src/dotnetcore/scriptgenerator.go

```diff
@@ -33,132 +35,120 @@ func (gen *DotnetCoreStartupScriptGenerator) GenerateEntrypointScript(scriptBuil
-		# Starting ASP.NET Core 3.0, an executable is created based on the platform where it is published from,
-		# so for example, if a user does a publish (not self-contained) on Mac, there would be files like 'todoApp'
-		# and 'todoApp.dll'. In this scenario the 'todoApp' executable is actually meant for Mac and not for Linux.
-		# So here we check for the file type and fall back to using 'dotnet todoApp.dll'.
-
-		isLinuxExecutable $startupExecutableFileName
-		if [ "$isLinuxExecutableResult" == "true" ]; then
-			echo "Found the startup executable file '$startupExecutableFileName'"
-			startUpCommand="./$startupExecutableFileName"
-		else
-			echo "Cannot use executable '$startupExecutableFileName' as startup file as it is not meant for Linux"
-		fi
-	fi
-
-	if [ -z "$startUpCommand" ] && [ ! -z "$startupDllFileName" ]; then
-		if [ -f "$startupDllFileName" ]; then
-			echo "Found the startup file '$startupDllFileName'"
-			startUpCommand="dotnet '$startupDllFileName'"
-		else
-			echo "Cound not find the startup dll file '$startupDllFileName'"
-		fi
-	fi
-fi
-
-if [ -z "$startUpCommand" ]; then
-	echo "Trying to run the default app instead..."
-	if [ ! -z "$defaultAppFilePath" ]; then
-		if [ -f "$defaultAppFilePath" ]; then
-			cd "$defaultAppFileDir"
-			startUpCommand="dotnet '$defaultAppFilePath'"
-		else
-			echo "Could not find the default app file '$defaultAppFilePath'"
-		fi
-	else
-		echo "Default app was not provided. Unable to start the application."
-	fi
-fi
-
-if [ -z "$startUpCommand" ]; then
-	exit 1
-fi
-
-echo "Running the command '$startUpCommand'..."
-eval "$startUpCommand"
-`
-	scriptBuilder.WriteString(script)
 	var runScript = scriptBuilder.String()
 	logger.LogInformation("Run script content:\n" + runScript)
 	return runScript
 }
+
+func (gen *DotnetCoreStartupScriptGenerator) getRuntimeConfigJsonFiles() []string {
+	logger := common.GetLogger("dotnetcore.scriptgenerator.getRuntimeConfigJsonFiles")
+	defer logger.Shutdown()
+
+	fileList := make([]string, 0)
+
+	appDir, err := os.Open(gen.AppPath)
+	if err != nil {
+		fmt.Printf(
+			"An error occurred while trying to look for '%s' files under '%s'.\n",
+			RuntimeConfigJsonExtension,
+			appDir.Name())
+		logger.LogError("An error occurred while trying to look for '%s' files under '%s'. Exception: '%s'",
+			RuntimeConfigJsonExtension,
+			appDir.Name(),
+			err)
+		return fileList
+	}
+	defer appDir.Close()
+
+	files, err := appDir.Readdir(-1)
+	if err != nil {
+		fmt.Printf(
+			"An error occurred while trying to look for '%s' files under '%s'.\n",
+			RuntimeConfigJsonExtension,
+			appDir.Name())
+		logger.LogError("An error occurred while trying to look for '%s' files under '%s'. Exception: '%s'",
+			RuntimeConfigJsonExtension,
+			appDir.Name(),
+			err)
+		return fileList
+	}
+
+	for _, file := range files {
+		if file.Mode().IsRegular() {
+			fileName := file.Name()
+			if strings.HasSuffix(fileName, RuntimeConfigJsonExtension) {
```

Comment:
should we also check that the corresponding dll file really exists?

Reply:
yeah, I agree...will update